core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -114,9 +115,11 @@ public void writeTo(StreamOutput out) throws IOException {
     private ShardId shardId;
     private String id;
     private String type;
+    private String input;         //added
     private long version;
     private long seqNo;
     private boolean forcedRefresh;
+
     protected Result result;
 
     public DocWriteResponse(ShardId shardId, String type, String id, long seqNo, long version, Result result) {
```

Comment:
We try to avoid creating tons of these constructors. Since this ctor is only called internally it should be fine to add the argument. Though, to be honest, it might make sense to do that in a followup PR because what you've done keeps this PR nice and readable. If you want to do it that way I'd leave a comment like:
```
// NORELEASE drop this ctor in favor of the new one below
```

That way even if you forget to make the followup someone else will do it. You'd have to explain your reasoning for adding the second ctor in the PR, but I think that reasoning is sound.